zk/stages/stage_sequence_execute.go

```diff
@@ -394,17 +373,39 @@ Loop:
 			continue
 		}
 
+		//[zkevm] - get the last batch number so we can check for empty batches in between it and the new one
+		lastBatchInserted, err := hermezDb.GetBatchNoByL2Block(stageProgress - 1)
+		if err != nil {
+			return fmt.Errorf("failed to get last batch inserted: %v", err)
+		}
+
+		/*
+		* SEQ: here we have our candidate batch
+		 */
+		// write batches between last block and this if they exist
+		currentBatch, err := hermezDb.GetBatchNoByL2Block(blockNum)
+		if err != nil {
+			return err
+		}
+
+		/*
+		* SEQ: here in theory we should get GERs from the L1 and write them down before
+		* execution into a separate table.
+		 */
```

Comment:
Could this be its own stage entirely?  To just keep a watch on GER events from the L1 so they're always ready ahead of time?

Reply:
yeah, good point, we need some L1 GER stage